osquery/worker/ipc/linux/linux_table_container_ipc.cpp

```diff
@@ -0,0 +1,450 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include "linux_table_container_ipc.h"
+
+#include <fcntl.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <syscall.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include <chrono>
+#include <cstdlib>
+#include <iostream>
+
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/worker/ipc/posix/pipe_channel_factory.h>
+#include <osquery/worker/ipc/table_ipc_json_converter.h>
+
+#include "osquery/worker/logging/glog/glog_logger.h"
+
+namespace osquery {
+
+DECLARE_bool(verbose);
+
+CLI_FLAG(bool,
+         keep_container_worker_open,
+         false,
+         "Keep the container worker running to be reused instead of closing it "
+         "after each query");
+
+namespace {
+
+const std::string kProc = "/proc";
+const std::string kMountNamespace = "/ns/mnt";
+
+PlatformProcess current_running_process;
+
+Status extractMountNamespaceId(const std::string& mount_namespace_path,
+                               std::string& mount_namespace_id) {
+  std::string mnt_namespace_id_source(33, 0);
```

Comment:
Why 33? Do we know this is always enough?

Reply:
Ah, good catch; so this is somewhat arbitrary, it comes from stracing `ls -l` against `/proc/<pid>/ns/mnt`.
Basically it tries multiple times to `readlink` until the number of characters read in two occasions is the same.
This happens when it arrives at 32 characters, so I took 33 (because I forgot that readlink does not try to append any null byte),
What I'm reading from that path though is something like `mnt:[4026531840]`; the id, which unfortunately doesn't seem to have definitions accessible from user space, is an unsigned integer so all together 16 characters should do (and I should change it).
Makes sense?

Reply:
That should probably be in a comment.

Reply:
yep, did that in this commit https://github.com/osquery/osquery/pull/6209/commits/f5bec4fd795eab203326665b515af19121530182